src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/interceptor/secure/WfsRequestInterceptor.java

```diff
@@ -0,0 +1,247 @@
+        LOG.debug("Intercepting WFS Transaction");
+        return isAllowed(request, "UPDATE") ? new GeoserverAuthHeaderRequest(
+            request, gsUser, gsPass) : forbidRequest(request);
+    }
+
+    /**
+     * Check for permission on layer by getting it through service.
+     * For update permissions, we check through custom method.
+     *
+     * @param request   The request
+     * @param paramName The optional parameter name for the layer parameter
+     * @param method    The access method like read or write
+     * @return if the layer is allowed to read for current user
+     */
+    private boolean isAllowed(
+        MutableHttpServletRequest request, String paramName, String method) {
+        String typeNameParam = request.getParameterIgnoreCase(paramName);
+        List<Layer> all = layerService.findAll();
+        boolean match = false;
+        for (Layer layer : all) {
+            if (layer.getSource() instanceof WfsLayerDataSource) {
+                WfsLayerDataSource source = (WfsLayerDataSource)
+                    layer.getSource();
+                if ((source.getTypeName().equalsIgnoreCase(typeNameParam) ||
+                     source.getTypeNames().equalsIgnoreCase(typeNameParam)) &&
+                    source.getUrl().equalsIgnoreCase(request.getContextPath() +
+                    "/geoserver.action")) {
+                    if (method.equals("UPDATE")) {
+                        match = checkForPermission(layer, Permission.UPDATE);
+                    } else if (method.equals("READ")) {
+                        // implicitly checked by findAll method
+                        match = true;
+                    }
+                }
+            } else if (layer.getSource() instanceof ImageWmsLayerDataSource) {
+                ImageWmsLayerDataSource source = (ImageWmsLayerDataSource)
+                    layer.getSource();
+                if (source.getLayerNames().equalsIgnoreCase(typeNameParam) &&
+                    source.getUrl().equalsIgnoreCase(request.getContextPath() +
+                    "/geoserver.action")) {
+                    if (method.equals("UPDATE")) {
+                        match = checkForPermission(layer, Permission.UPDATE);
+                    } else if (method.equals("READ")) {
+                        // implicitly checked by findAll method
+                        match = true;
+                    }
+                }
+            }
+        }
+        return match;
+    }
+
+    /**
+     * Calls main method with default "typename" parameter
+     *
+     * @param request The request
+     * @param method The access method like read or write
+     * @return If the layer is allowed to read for current user
+     */
+    private boolean isAllowed(MutableHttpServletRequest request, String method) {
+        String typeNameParam = "typeNames";
+        if (StringUtils.isEmpty(request.getParameterIgnoreCase(typeNameParam))) {
+            typeNameParam = "typeName";
```

Comment:
You might want to check if it's possible to reuse the [enums](https://github.com/terrestris/shogun-core/blob/master/src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/enumeration/OgcEnum.java#L252) here.